### PR TITLE
Rcal 98 tweak core metadata

### PR DESCRIPTION
**Pull request by PaulHuwe**
Tweaked metadata to reconcile with INS sample files. Changes included:
Removed optical_element from required metadata, added MASK to reftype enum.

### CODE REVIEW

File: romancal/datamodels/schemas/core.schema.yaml [hunk 1]
```diff
@@ -328,7 +328,7 @@ properties:
             title: Name of the filter element used
             type: string
             enum: [F062, F087, F106, F129, W146, F158, F184, GRISM, PRISM, DARK, ENGINEERING]
-        required: [detector, optical_element, name]
```
Comment (eslavich) [inline romancal/datamodels/schemas/core.schema.yaml:331]: This field is still required in the flat schema, right?  Should we make it required there and in other schemas that need it?

Comment (PaulHuwe) [reply romancal/datamodels/schemas/core.schema.yaml:331]: I have sent a mail to TPTB about how best to implement that for subsets of models that require them.

Comment (PaulHuwe) [reply romancal/datamodels/schemas/core.schema.yaml:331]: TPTB = The Powers That Be aka @ddavis-stsci, @nden, and @sosey

**Review by sosey — CHANGES_REQUESTED** [review]

File: romancal/datamodels/schemas/core.schema.yaml (hunk 1 above)
Comment (PaulHuwe) [reply romancal/datamodels/schemas/core.schema.yaml:331]: @sosey, Did you get a chance to see the mail I sent you three last week where I suggested making "requirement profile" schema files that could then be included by appropriate schema files? Doing so could eliminate having to duplicate requirements across many schema files.

File: romancal/datamodels/schemas/reference_files/referencefile.schema.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ allOf:
         reftype:
           title: Reference file type
           type: string
-          enum: [AREA, DARK, FLAT, GAIN, IPC, LINEARITY, PERSAT, PHOTOM, READNOISE, REFPIX,
+          enum: [AREA, DARK, FLAT, GAIN, IPC, LINEARITY, MASK, PERSAT, PHOTOM, READNOISE, REFPIX,
```
[the remark this answers is not in the document]
Comment (PaulHuwe) [reply romancal/datamodels/schemas/reference_files/referencefile.schema.yaml:16]: I will remove that.

[the remark this answers is not in the document]
Comment (PaulHuwe) [reply romancal/datamodels/schemas/reference_files/referencefile.schema.yaml]: Done.

File: romancal/datamodels/schemas/core.schema.yaml (hunk 1 above)
Comment (PaulHuwe) [reply romancal/datamodels/schemas/core.schema.yaml:331]: Taken to RCAL-126

**Review by ddavis-stsci — APPROVED** [review]
LGTM

**Review by nden — APPROVED** [review]

File: romancal/datamodels/schemas/core.schema.yaml [hunk 3]
```diff
@@ -328,7 +328,7 @@ properties:
             title: Name of the filter element used
             type: string
             enum: [F062, F087, F106, F129, W146, F158, F184, F213, GRISM, PRISM, DARK]
-        required: [detector, optical_element, name]
```
Comment (nden) [inline romancal/datamodels/schemas/core.schema.yaml:331]: Just curious why `optical_element` is not required. What kind of files are we getting that may not have it?

Comment (PaulHuwe) [reply romancal/datamodels/schemas/core.schema.yaml:331]: Some reference files (gain is an example).

**Pull request merged**